BibBuddy/app/src/main/java/de/bibbuddy/BookNotesView.java

```diff
@@ -190,33 +186,31 @@ private void handleDeleteNote() {
     alertDeleteBookNote.setNegativeButton(R.string.back, new DialogInterface.OnClickListener() {
       @Override
       public void onClick(DialogInterface dialog, int which) {
+        deselectNoteItems();
       }
     });
 
     alertDeleteBookNote.setPositiveButton(R.string.delete, new DialogInterface.OnClickListener() {
       @Override
       public void onClick(DialogInterface dialog, int which) {
-        final int notesNumber = selectedNoteItems.size();
-
-        bookNotesViewModel.deleteNotes(selectedNoteItems);
-        adapter.notifyDataSetChanged();
-        updateBookNoteList(bookNotesViewModel.getCurrentNoteList());
-        updateEmptyView(bookNotesViewModel.getCurrentNoteList());
-
+        final int notesNumber = adapter.getSelectedNoteItems().size();
         if (notesNumber > 1) {
           Toast.makeText(context, getString(R.string.deleted_notes), Toast.LENGTH_SHORT).show();
         } else {
           Toast.makeText(context, getString(R.string.deleted_note), Toast.LENGTH_SHORT).show();
         }
-
-        unselectNoteItems();
+        bookNotesViewModel.deleteNotes(adapter.getSelectedNoteItems());
```

Comment:
Is it logical to drop the Toast before Notes are actually deleted? Should be the other way round I guess? In all other classes where items are deleted we did it this way :)

Reply:
I did that, because Lint told me not to put so many lines between the variable instantiation and its usage (since you have to fetch the selected item number before deleting items). I still reverted that change and the toast is displayed at the end of that code-part.

Reply:
Thanks for the reply - my bad! Then of course the solution approved by checkstyle is the better one :)